🚩: Milestone 2: Vet Clinic | feature/queryAndUpdateAnimals

Ruthmy opened the pull request:
## 🚩: Milestone 2: Vet Clinic | feature/queryAndUpdateAnimals - >>branch<< -

### Here is a summary of what has been done:

- Follow [Gitflow Workflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md) in this repository.

- Added the collumn `species` type string (`varchar(50)`).
- Inserted 7 rows in the table `animals` with the provided data.
- Executed 3 transactions on the database, to added the values of the collumn `species` in the table `animals`, to delete all records, to delete all animals born after 2022 and change the weight from negative to positive.
- Added the queries to ask the provided questions.
- Added in the `queries.sql` the queries required in the assignment.

---

Screenshot of the queries performed:

<img src="https://raw.githubusercontent.com/Ruthmy/vet-clinic-db/0f97c92d003302024ab60b04d04e41fb87fe3c93/assests/2-task/1-query.png">
<br>
<img src="https://raw.githubusercontent.com/Ruthmy/vet-clinic-db/0f97c92d003302024ab60b04d04e41fb87fe3c93/assests/2-task/2-query.png">
<br>
<img src="https://raw.githubusercontent.com/Ruthmy/vet-clinic-db/0f97c92d003302024ab60b04d04e41fb87fe3c93/assests/2-task/3-query.png">
<br>
<img src="https://raw.githubusercontent.com/Ruthmy/vet-clinic-db/0f97c92d003302024ab60b04d04e41fb87fe3c93/assests/2-task/4-query.png">
<br>
<img src="https://raw.githubusercontent.com/Ruthmy/vet-clinic-db/0f97c92d003302024ab60b04d04e41fb87fe3c93/assests/2-task/5-query.png">
<br>

---

### Thank you for taking the time to review this PR ⭐!

---

Ruthmy committed:
Modified 'schema.sql'
- Removed the check from the weight column in order to complete the assigned task.

Ruthmy committed:
Modified 'data.sql'
- The species column has been created and the necessary information has been inserted.

Ruthmy committed:
Modified 'queries.sql'
- Executed the necessary transactions to the uploaded information.

Ruthmy committed:
Modified 'schema.sql'
Added the column 'species'

Ruthmy committed:
Modified 'data.sql'
Fixed a typo in the escape_attemps column name

Ruthmy committed:
Modified 'queries.sql'
- Added the new queries to anwser the provided questions.
- Fixed a typo with the scape_attemps column name.

Ruthmy committed:
Created 'assests' -folder-
- Added all the images of the performed queries.

LionRouge1 commented on data.sql:
```diff
@@ -1,6 +1,18 @@
+
+/* From task: query and update animals table */
+
+ALTER TABLE animals ADD COLUMN species VARCHAR(50) NOT NULL DEFAULT 'unknown';
```
- [x] Kindly note that everything that creates or changes the structure of the tables should be saved in the `schema.sql` file. Kindly move this code to the `schema.sql`. Because it changes the structure of the table `animals`

Ruthmy replied:
Hi @LionRouge1 
You're totally right, I forgot to separate it, I will make the change.
Thank u very much

LionRouge1 commented on queries.sql:
```diff
@@ -2,9 +2,77 @@
+/* From task: query and update animals table */
+
+-- Inside a transaction update animals table by setting the species column to unspecified. Verify change. roll back the change and verify again.
+BEGIN TRANSACTION;
+
+UPDATE animals SET species = 'unspecified';
+SELECT * FROM animals;
+
+ROLLBACK;
```
- [x] Nice work setting the transaction :clap: However for this step, it required verifying that the change was made and rollbacked, therefore I suggest you verify that the change was undone after the `ROLLBACK` command.
```sql
   BEGIN; -- start transaction

   UPDATE animals SET species = 'unspecified'; -- make change

   SELECT * from animals; -- verify that change was made
   ROLLBACK;

   SELECT species from animals; -- verify that change was undone
```

Ruthmy replied:
Hi again @LionRouge1 !!
Thanks for your feedback. You're right, I executed the statement, but did not add it to the file. The change is done.
Thank you very much for your time and review. ✨

LionRouge1 reviewed (changes requested):
Hi @Ruthmy, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

![almost](https://media.giphy.com/media/KyBesf8kNT6s88SEgb/giphy.gif)

## Highlights
- Use transactions to update and delete records :heavy_check_mark: 
- Use aggregate functions and `GROUP BY` to answer analytical questions :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Ruthmy committed:
Modified 'data.sql' and 'schema.sql'
- Moved to the corresponding file the statement that adds the new species column.

Ruthmy committed:
Modified 'queries.sql'
- Added the missing sentence to verify that the changes were not applied.

abdulhamiid reviewed (approved):
# :green_circle:  Approved! :partying_face:
Hi @Ruthmy 👋, 
![image](https://user-images.githubusercontent.com/55596970/158098566-50966056-ce3d-4c4b-aac3-cce7a7d107fb.png)
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 🌟

- ✔️ Descriptions are very descriptive
- ✔️ Use database transactions.
- ✔️ Modify and delete data in SQL.

## Optional suggestions 🤷‍♂️
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

## Important Notes 📝
- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
